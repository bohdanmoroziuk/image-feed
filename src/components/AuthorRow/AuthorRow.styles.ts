import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    height: 50,
    paddingHorizontal: 10,
  },
  text: {
    flex: 1,
    marginHorizontal: 6,
  },
});
