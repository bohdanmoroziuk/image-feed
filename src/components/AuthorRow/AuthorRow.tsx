import { FunctionComponent } from 'react';
import { View, Text, TouchableOpacity } from 'react-native';

import getInitials from 'src/utils/getInitials';
import getAvatarColor from 'src/utils/getAvatarColor';

import Avatar from 'src/components/Avatar';

import styles from './AuthorRow.styles'

export interface AuthorRowProps {
  fullName: string;
  linkText?: string;
  onPressLinkText?: () => void;
}

export const AuthorRow: FunctionComponent<AuthorRowProps> = ({
  fullName,
  linkText,
  onPressLinkText,
}) => {

  return (
    <View style={styles.container}>
      <Avatar
        size={35}
        color={getAvatarColor(fullName)}
        initials={getInitials(fullName)}
      />
      <Text
        style={styles.text}
        numberOfLines={1}
      >
        {fullName}
      </Text>
      {!!linkText && (
        <TouchableOpacity onPress={onPressLinkText}>
          <Text numberOfLines={1}>
            {linkText}
          </Text>
        </TouchableOpacity>
      )}
    </View>
  );
};
