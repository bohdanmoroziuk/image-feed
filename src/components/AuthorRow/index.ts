import { AuthorRow } from './AuthorRow';

export type { AuthorRowProps } from './AuthorRow';
export default AuthorRow;
