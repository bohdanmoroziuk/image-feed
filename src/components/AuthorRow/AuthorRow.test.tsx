import { render, fireEvent } from '@testing-library/react-native';

import { AuthorRow } from './AuthorRow';

describe('AuthorRow', () => {
  it('renders successfully', () => {
    const { toJSON } = render(
      <AuthorRow
        fullName="First Last"
        linkText="Comments"
        onPressLinkText={() => {}}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const { getByText } = render(
      <AuthorRow
        fullName="First Last"
        linkText="Comments"
        onPressLinkText={() => {}}
      />
    );

    expect(getByText('FL')).toBeDefined();
    expect(getByText('Comments')).toBeDefined();
  });

  it('calls the onPressLinkText handler when the link text is pressed', () => {
    const mockOnPressLinkText = jest.fn();

    const { getByText } = render(
      <AuthorRow
        fullName="First Last"
        linkText="Comments"
        onPressLinkText={mockOnPressLinkText}
      />
    );

    fireEvent.press(getByText('Comments'));

    expect(mockOnPressLinkText).toHaveBeenCalled();
  })
});
