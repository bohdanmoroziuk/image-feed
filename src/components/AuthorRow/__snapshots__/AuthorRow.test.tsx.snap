// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorRow renders successfully 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 50,
      "paddingHorizontal": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#0b0b9e",
          "borderRadius": 17.5,
          "height": 35,
          "width": 35,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      FL
    </Text>
  </View>
  <Text
    numberOfLines={1}
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 6,
      }
    }
  >
    First Last
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      numberOfLines={1}
    >
      Comments
    </Text>
  </View>
</View>
`;
